Export Wireshark CLI tools from formula

luizberti opened the pull request:
Wireshark comes with several bundled CLI tools, which currently are not
exposed through the cask, and people have to manually alias or link them.

This makes the Wireshark cask work more in line with what other casks
that bundle CLI binaries (e.g. `julia` or `visual-studio-code`) do.

The following command was used to extract the available binaries from
`Wireguard.app` if needed for future reference (needs `fd` and `rg`):

```bash
fd --type=executable . /Applications/Wireshark.app/ | \
rg -F Wireshark.app/Contents/MacOS | rg -v 'Wireshark$' | \
rg -F /Applications --replace='binary "#{appdir}' | awk '{print $0 "\""}'
```

The outputs of it are in line with the documentation currently hosted at:
https://www.wireshark.org/docs/man-pages/

A great follow up addition to this would be adding the `man` pages (e.g.
`/Applications/Wireshark.app/Contents/Resources/share/man/man1/tshark.1`)
automatically to `man`'s path through the Cask formula, but I am not sure
how to accomplish this **(I'll gladly make another PR  for this if a maintainer
can point me to an example from another formula that does this)**.

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [ ] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

luizberti commented:
**For the reviewers:** I fixed some style errors that were already present by running the `brew cask style --fix` command, except for the `Missing frozen string literal comment` one (cause I'm not exactly sure what that's about, since I'm not a rubyist)

The only style violation that was actually introduced by my PR was the `Block has too many lines` one, but I am not exactly sure what to do about it given that all those lines are necessary, so some guidance would be appreciated here.

miccal commented:
Thank you @luizberti, but I believe the CLI tools can be installed via the [`wireshark` formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/wireshark.rb). Note that the `caveats` for the formula state that:
```
==> Caveats
This formula only installs the command-line utilities by default.

Install Wireshark.app with Homebrew Cask:
  brew cask install wireshark

If your list of available capture interfaces is empty
(default macOS behavior), install ChmodBPF:
  brew cask install wireshark-chmodbpf
```
&nbsp;
@vitorgalvao I would suggest that we add a similar `caveat` message to the cask version to match the formula.

luizberti commented:
@miccal @vitorgalvao Fair enough, I removed the linkage statements in favor of a caveat message to avoid confusion, can you take a look at it again?

luizberti commented:
Actually, scratch what I just said. I just tried installing the `wireshark` formula, and it skips linking the binaries if the cask is installed, because it assumes the cask does the linking, and indeed none of the CLI tools (e.g. `tshark` and `randpkt`) are available after installing the formula. See the output of `brew install wireshark`:
```
...
==> Installing wireshark dependency: nghttp2
==> Pouring nghttp2-1.41.0_1.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/nghttp2/1.41.0_1: 26 files, 2.7MB
==> Installing wireshark
==> Pouring wireshark-3.2.7.catalina.bottle.tar.gz
==> wireshark cask is installed, skipping link.                  <<========= HERES THE RELEVANT PART
==> Caveats
This formula only installs the command-line utilities by default.

Install Wireshark.app with Homebrew Cask:
  brew cask install wireshark

If your list of available capture interfaces is empty
(default macOS behavior), install ChmodBPF:
  brew cask install wireshark-chmodbpf
==> Summary
🍺  /usr/local/Cellar/wireshark/3.2.7: 1,168 files, 91.3MB
==> Caveats
==> wireshark
This formula only installs the command-line utilities by default.
...
```

I really do think linking from the cask is what makes the most sense, as it keeps the CLI tools and Wireshark.app versioned and updated together, and given that this seems to be the behavior the regular `wireshark` brew formula expects, I think I should revert my commit that removed the symlinking instructions and added the caveat.

Do the two of you agree? @miccal @vitorgalvao

vitorgalvao commented:
> I really do think linking from the cask is what makes the most sense, as it keeps the CLI tools and Wireshark.app versioned and updated together, and given that this seems to be the behavior the regular `wireshark` brew formula expects, I think I should revert my commit that removed the symlinking instructions and added the caveat.

Agreed. That’s consistent with other casks ([`mpv`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/mpv.rb), [`mkvtoolnix`](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/mkvtoolnix.rb)).

luizberti commented:
@miccal @vitorgalvao the commit has been reverted back to linking all the binaries from the cask's app

anything else pending for a merge?

miccal commented:
Thank you @luizberti.
